openapi_python_client/parser/properties/merge_properties.py

```diff
@@ -27,7 +34,12 @@
 STRING_WITH_FORMAT_TYPES = (DateProperty, DateTimeProperty, FileProperty)
 
 
-def merge_properties(prop1: Property, prop2: Property) -> Property | PropertyError:  # noqa: PLR0911
+def merge_properties(  # noqa:PLR0911
+    prop1: Property,
+    prop2: Property,
+    parent_name: str,
+    config: Config,
```

Comment:
We need `parent_name` and `config` now because it might be necessary to generate a new inline schema.